neuralop/layers/spectral_convolution.py

```diff
@@ -482,24 +492,22 @@ def forward(
             slices_x[-1] = slice(None, weight.shape[-1])
         else:
             slices_x[-1] = slice(None)
-        
         out_fft[slices_x] = self._contract(x[slices_x], weight, separable=self.separable)
-
         if self.resolution_scaling_factor is not None and output_shape is None:
             mode_sizes = tuple([round(s * r) for (s, r) in zip(mode_sizes, self.resolution_scaling_factor)])
 
         if output_shape is not None:
             mode_sizes = output_shape
-
+        
         if self.order > 1:
             out_fft = torch.fft.fftshift(out_fft, dim=fft_dims[:-1])
         
         if self.complex_data:
-            x = torch.fft.ifftn(out_fft, s=mode_sizes, dim=fft_dims, norm=self.fft_norm)
+            out_fft = torch.fft.ifftn(out_fft, s=mode_sizes, dim=fft_dims, norm=self.fft_norm)
         else:
-            x = torch.fft.irfftn(out_fft, s=mode_sizes, dim=fft_dims, norm=self.fft_norm)
+            out_fft = torch.fft.irfftn(out_fft, s=mode_sizes, dim=fft_dims, norm=self.fft_norm)
```

Comment:
Why the change here? I don't think it should change anything for memory usage since we're not doing the operation inplace. We can always call x.add_ below if needed.